test/cql-pytest/test_wasm.py

```diff
@@ -0,0 +1,396 @@
+# -*- coding: utf-8 -*-
+# Copyright 2021-present ScyllaDB
+#
+# This file is part of Scylla.
+#
+# Scylla is free software: you can redistribute it and/or modify
+# it under the terms of the GNU Affero General Public License as published by
+# the Free Software Foundation, either version 3 of the License, or
+# (at your option) any later version.
+#
+# Scylla is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU Affero General Public License
+# along with Scylla.  If not, see <http://www.gnu.org/licenses/>.
+
+#############################################################################
+# Tests for user defined functions defined with WebAssembly backend
+#############################################################################
+
+from cassandra.protocol import InvalidRequest
+from util import new_test_table, unique_name
+
+import pytest
+
+# Test that calling a wasm-based fibonacci function works
+def test_fib(cql, test_keyspace):
```

Comment:
All of the tests in this file are Scylla-only (they can't work on Cassandra because it doesn't support wasm, right?). So we should either add a "scylla-only" on each test here, or, simpler, decide that all functions in this file are Scylla only - perhaps with an auto-use fixture like this (please test, I didn't):
```
# WASM support is Scylla-only (it's not supported in Cassandra),
# so all tests in this file are Scyla-only
@pytest.fixture(scope="module", autouse=True)
def all_tests_here_are_scylla_only(scylla_only):
    yield
```

Reply:
Another important point: all the tests here will fail if Scylla is not compiled with WASM support (which if I understood correctly is optional). So we should skip all these tests not just on Cassandra, but in any case where WASM is not supported. So that fixture mentioned above shouldn't use scylla_only, and instead should check if some basic WASM request works, and if it doesn't, skip.

Reply:
Yes, the tests are definitely Scylla-only, I forgot about this fixture. I'll also add some logic for skipping these tests if the wasm engine was not embedded during compilation.